src/ripple/app/tx/impl/AMMBid.cpp

```diff
@@ -204,10 +207,10 @@ applyBid(
                           Number const& burn) -> TER {
         auctionSlot.setAccountID(sfAccount, account_);
         auctionSlot.setFieldU32(sfExpiration, current + TOTAL_TIME_SLOT_SECS);
-        if (fee == 0)
-            auctionSlot.makeFieldAbsent(sfDiscountedFee);
-        else
+        if (fee != 0)
             auctionSlot.setFieldU16(sfDiscountedFee, fee);
+        else if (auctionSlot.isFieldPresent(sfDiscountedFee))
+            auctionSlot.makeFieldAbsent(sfDiscountedFee);
```

Comment:
I also prefer to check if it exists first. There is an interface inconsistency between STObject and the Ledger. If you try to delete an SLE that doesn't exist, it's a logic error. If you try to delete a field that doesn't exist it ignores you. Just because someone made a mistake programming the APIs doesn't mean programmers should make a mistake using them.

Reply:
If you think of the more modern API, which uses `std::optional`, then `makeFieldAbsent` is kind of like assigning `std::nullopt` to a `std::optional`, which is fine whether the `std::optional` is seated or not.

I don’t mind checking, but I don’t see a problem with not checking.

Reply:
It'll throw `fieldNotFound` if it doesn't exist. It's a default field. I can check whether it has the default value. It's still a check. Try this test:
>     try
>        {
>           auto sle = std::make_shared<SLE>(keylet::amm(xrpIssue(), USD));
>            auto& auctionSlot = sle->peekFieldObject(sfAuctionSlot);
>            auctionSlot.makeFieldAbsent(sfTradingFee);
>        } catch (std::exception const& e)
>        {
>            std::cout << "exception " << e.what() << std::endl;
>        }